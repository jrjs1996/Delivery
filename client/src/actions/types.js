export const FETCH_CUSTOMERS = 'FETCH_CUSTOMERS';
export const ADD_CUSTOMER = 'ADD_CUSTOMER';

export const ADD_ORDER = 'ADD_ORDER';
export const FETCH_ORDERS = 'FETCH_ORDER';
export const COMPLETE_ORDER = 'COMPLETE_ORDER';
export const GET_CUSTOMER = 'GET_CUSTOMER';

export const LOGIN_ADMIN = 'LOGIN_ADMIN';
export const FETCH_ADMINS = 'FETCH_ADMINS';

export const FETCH_MENU = 'FETCH_MENU';
export const ADD_MENU_ITEM = 'ADD_MENU_ITEM';
export const UPDATE_MENU_ITEM = 'UPDATE_MENU_ITEM';
export const DELETE_MENU_ITEM = 'DELETE_MENU_ITEM';
