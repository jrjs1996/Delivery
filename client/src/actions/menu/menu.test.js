import configureStore from 'redux-mock-store';
import * as menuActions from './menu';
import * as actionTypes from '../types';
import mockAxios from '../../../__mocks__/axios';

const mockStore = configureStore();
const store = mockStore();

describe('Add menu item', () => {
  beforeEach(() => {
    store.clearActions();
  });

  it('Dispatches ADD_MENU_ITEM with payload of response data on success', async () => {
    await menuActions.addMenuItem({ mockPostData: 'MockPostData' })(store.dispatch);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('Dispatches ERROR on error', async () => {
    mockAxios.giveError(mockAxios.post);
    await menuActions.addMenuItem({ mockPostData: 'MockPostData' })(store.dispatch);
    expect(store.getActions()).toMatchSnapshot();
  });
});

describe('Delete menu item', () => {
  beforeEach(() => {
    store.clearActions();
  });

  it('Dispatches DELETE_MENU item with payload of given menu item number', async () => {
    await menuActions.deleteMenuItem(1234)(store.dispatch);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('Dispatches ERROR on error', async () => {
    mockAxios.giveError(mockAxios.delete);
    await menuActions.deleteMenuItem(1234)(store.dispatch);
    expect(store.getActions()).toMatchSnapshot();
  });
});

describe('Fetch menu', () => {
  beforeEach(() => {
    store.clearActions();
  });

  it('Dispatches FETCH_MENU with payload of response data', async () => {
    await menuActions.fetchMenu()(store.dispatch);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('Dispatches ERROR on error', async () => {
    mockAxios.giveError(mockAxios.get);
    await menuActions.fetchMenu()(store.dispatch);
    expect(store.getActions()).toMatchSnapshot();
  });
});

describe('Update menu item', () => {
  beforeEach(() => {
    store.clearActions();
  });

  it('Dispatches UPDATE_MENU_ITEM with payload of response data', async () => {
    await menuActions.updateMenuItem({ mockPutData: 'MockPutData' })(store.dispatch);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('Dispatches ERROR on error', async () => {
    mockAxios.giveError(mockAxios.put);
    await menuActions.updateMenuItem({ mockPutData: 'MockPutData' })(store.dispatch);
    expect(store.getActions()).toMatchSnapshot();
  });
});

describe('Upload menu item image', () => {
  beforeEach(() => {
    store.clearActions();
  });

  it('Dispatches UPDATE_MENU_ITEM with payload of response data', async () => {
    await menuActions.uploadMenuItemImage(1234, { mockFileInfo: 'MockFileInfo' })(store.dispatch);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('Dispatches ERROR on error', async () => {
    mockAxios.giveError(mockAxios.post);
    await menuActions.uploadMenuItemImage(1234, { mockFileInfo: 'MockFileInfo' })(store.dispatch);
    expect(store.getActions()).toMatchSnapshot();
  });
});
