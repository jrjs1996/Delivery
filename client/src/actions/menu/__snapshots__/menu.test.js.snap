// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add menu item Dispatches ADD_MENU_ITEM with payload of response data on success 1`] = `
Array [
  Object {
    "payload": Object {
      "mock": "MockData",
    },
    "type": "ADD_MENU_ITEM",
  },
]
`;

exports[`Add menu item Dispatches ERROR on error 1`] = `
Array [
  Object {
    "payload": "Unable to add menu item",
    "type": "ERROR",
  },
]
`;

exports[`Delete menu item Dispatches DELETE_MENU item with payload of given menu item number 1`] = `
Array [
  Object {
    "payload": Object {
      "menuNumber": 1234,
    },
    "type": "DELETE_MENU_ITEM",
  },
]
`;

exports[`Delete menu item Dispatches ERROR on error 1`] = `
Array [
  Object {
    "payload": "Unable to delete menu item",
    "type": "ERROR",
  },
]
`;

exports[`Fetch menu Dispatches ERROR on error 1`] = `
Array [
  Object {
    "payload": "Unable to fetch menu",
    "type": "ERROR",
  },
]
`;

exports[`Fetch menu Dispatches FETCH_MENU with payload of response data 1`] = `
Array [
  Object {
    "payload": Object {
      "mock": "MockData",
    },
    "type": "FETCH_MENU",
  },
]
`;

exports[`Update menu item Dispatches ERROR on error 1`] = `
Array [
  Object {
    "payload": "Unable to update menu item",
    "type": "ERROR",
  },
]
`;

exports[`Update menu item Dispatches UPDATE_MENU_ITEM with payload of response data 1`] = `
Array [
  Object {
    "payload": Object {
      "mock": "MockData",
    },
    "type": "UPDATE_MENU_ITEM",
  },
]
`;

exports[`Upload menu item image Dispatches ERROR on error 1`] = `
Array [
  Object {
    "payload": "Unable to update menu item image.",
    "type": "ERROR",
  },
]
`;

exports[`Upload menu item image Dispatches UPDATE_MENU_ITEM with payload of response data 1`] = `
Array [
  Object {
    "payload": Object {
      "mock": "MockData",
    },
    "type": "UPDATE_MENU_ITEM",
  },
]
`;
